python/grizzly/grizzly/grizzly.py

```diff
@@ -5,6 +5,367 @@
 from lazy_op import LazyOpResult, to_weld_type
 from weld.weldobject import *
 
+class DataFrameWeldLoc:
+    def __init__(self, df):
+        self.df = df
+
+    def __getitem__(self, key):
+        if isinstance(key, SeriesWeld):
+            # We're going to assume that the first column in these dataframes
+            # is an index column. This assumption does not hold throughout grizzly,
+            # so we should fix that moving forward.
+            index_expr = grizzly_impl.get_field(self.df.expr, 0)
+            if self.df.is_pivot:
+                index_type, pivot_type, column_type = self.df.column_types
+                index_elem_type = index_type.elemType
+                index_expr_predicate = grizzly_impl.isin(index_expr, key.expr, index_elem_type)
+                return DataFrameWeldExpr(
+                    grizzly_impl.pivot_filter(
+                        self.df.expr,
+                        index_expr_predicate
+                    ),
+                    self.df.column_names,
+                    self.df.weld_type,
+                    is_pivot=True
+                )
+            # TODO : Need to implement for non-pivot tables
+        raise Exception("Cannot invoke getitem on an object that is not SeriesWeld")
+
+class WeldLocIndexer:
```

Comment:
Same comment as above

Reply:
Addressed.
We can probably merge the WeldLocIndexer and DataFrameWeldLoc classes. 
LocIndexer and DataFrameLoc were two types of objects I encountered in Pandas but they seem to be having the same functionality which is just selecting rows based on an index column.